viahtml/hooks/_headers.py

```diff
@@ -0,0 +1,91 @@
+"""Specific configuration for headers."""
+
+
+class Headers:
+    """Methods for manipulating the headers we accept and emit."""
+
+    BLOCKED = {
+        # h CSRF token header
+        "X-Csrf-Token",
+    }
+
+    BLOCKED_INBOUND = {
+        # Passing CF headers on to via causes "Error 1000: DNS points to
+        # prohibited IP" errors at CloudFlare, so we have to strip them out
+        # here.
+        "Cdn-Loop",
+        "Cf-Connecting-Ip",
+        "Cf-Ipcountry",
+        "Cf-Ray",
+        "Cf-Request-Id",
+        "Cf-Visitor",
+        # AWS headers
+        "X-Amzn-Trace-Id",
+    } | BLOCKED
+
+    BLOCKED_OUTBOUND = {
+        # Disable the Content-Security-Policy which blocks our embed
+        "Content-Security-Policy",
+        # Various headers added by `pywb` relating to archival
+        "X-Archive-Orig-Age",
+        "X-Archive-Orig-Cache-Control",
+        "X-Archive-Orig-Date",
+        "X-Archive-Orig-Etag",
+        "X-Archive-Orig-Expires",
+        "X-Archive-Orig-Last-Modified",
+        "X-Archive-Orig-Server",
+        "X-Archive-Orig-Vary",
+        "X-Cache",
+        "X-Archive-Orig-Content-Length",
+        "Memento-Datetime",
+        "Link",
+    } | BLOCKED
```

Comment:
:man_shrugging: Should make it pretty simple to manage block headers going forward